plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -665,14 +672,30 @@ private static Optional<Object> translateValue(Object trinoNativeValue, Type typ
             return Optional.of(trinoNativeValue);
         }
 
+        if (type instanceof DecimalType decimalType) {
+            if (decimalType.isShort()) {
+                return Optional.of(Decimal128.parse(Decimals.toString((long) trinoNativeValue, decimalType.getScale())));
+            }
+            return Optional.of(Decimal128.parse(Decimals.toString((Int128) trinoNativeValue, decimalType.getScale())));
+        }
+
         if (type instanceof ObjectIdType) {
             return Optional.of(new ObjectId(((Slice) trinoNativeValue).getBytes()));
         }
 
+        if (type instanceof CharType charType) {
+            Slice slice = padSpaces(((Slice) trinoNativeValue), charType);
+            return Optional.of(slice.toStringUtf8());
+        }
+
         if (type instanceof VarcharType) {
             return Optional.of(((Slice) trinoNativeValue).toStringUtf8());
         }
 
+        if (type instanceof VarbinaryType) {
+            return Optional.of(new Binary(((Slice) trinoNativeValue).getBytes()));
```

Comment:
Yes, You are right.

The `VarbinaryType` in Trino does not have a equivalent for the binary "sub-types" found in MongoDB. I think it would be best to forego the implementation of the `varbinary` filter pushdown in this scenario.